source/satellites.tidal_stripping.radius.King1962.F90

```diff
@@ -279,12 +280,11 @@ double precision function king1962Radius(self,node)
        tidalTensorMatrix               = tidalTensorComponents
        call tidalTensorMatrix%eigenSystem(tidalTensorEigenVectors,tidalTensorEigenValues)
        tidalTensorEigenValueComponents = tidalTensorEigenValues
-       tidalTensorEigenVectorComponents= tidalTensorEigenVectors
-       tidalFieldRadial                =-abs(tidalTensor%vectorProject(tidalTensorEigenVectorComponents(maxloc(tidalTensorEigenValueComponents,dim=1),:))) &
-            &                           *(                                                                                                                 &
-            &                             +kilo                                                                                                            &
-            &                             *gigaYear                                                                                                        &
-            &                             /megaParsec                                                                                                      &
+       tidalFieldRadial                =-maxval(tidalTensorEigenValueComponents) &
```

Comment:
Maybe we should add an extra check here to ensure that `maxval(tidalTensorEigenValueComponents)` is always positive - to avoid stripping for compressive tidal fields? In practice, I think it's physically very unlikely (in the types of system we consider) to have a tidal field that is compressive in all directions - but I think it is technically possible.

So, we could change this to:
```
       tidalFieldRadial                =-max(maxval(tidalTensorEigenValueComponents),0.0d0) &
            &                           *(                                                  &
            &                             +kilo                                             &
            &                             *gigaYear                                         &
            &                             /megaParsec                                       &
            &                            )**2
```

Reply:
Hi Andrew, yes, I think you are right. I did not do that at first because there is a check later

```
    ! If the tidal force is stretching (not compressing), compute the tidal radius.
    if     (                                                                          &
         &   frequencyAngular**2                                   > tidalFieldRadial &
         &  .and.                                                                     &
         &   massSatellite                                         >  0.0d0           &
         &  .and.                                                                     &
         &   self%galacticStructure_%massEnclosed(node,radiusZero) >= 0.0d0           &
         & ) then
```

Here it check whether the centrifugal force+tidal force is compressive. But actually if there is no tidal force, we should expect zero tidal mass loss. So besides the change you suggested, we might also need to change the above lines to

```
    ! If the tidal force is stretching (not compressing), compute the tidal radius.
    if     (                                                                          &
         &   tidalFieldRadial                                      >  0.0d0           &
         &  .and.                                                                     &
         &   massSatellite                                         >  0.0d0           &
         &  .and.                                                                     &
         &   self%galacticStructure_%massEnclosed(node,radiusZero) >= 0.0d0           &
         & ) then
```

Reply:
Good point. But, even if there is no tidal field, in principle the centrifugal force can lead to some stripping I think? If the centrifugal force in the outer parts of the subhalo exceeds the self-gravity of the subhalo then particles could be lost. If that's the case then I think it would be best to keep:
```
       tidalFieldRadial                =-maxval(tidalTensorEigenValueComponents) &
            &                           *(                                        &
            &                             +kilo                                  &
            &                             *gigaYear                              &
            &                             /megaParsec                            &
            &                            )**2
```
and then keep the original check for compressive forces, but maybe for simplicity and clarity change it to:
```
    ! If the tidal force is stretching (not compressing), compute the tidal radius.
    tidalPull=frequencyAngular**2-tidalFieldRadial
    if     (                                                                          &
         &   tidalPull                                             >  0.0d0           &
         &  .and.                                                                     &
         &   massSatellite                                         >  0.0d0           &
         &  .and.                                                                     &
         &   self%galacticStructure_%massEnclosed(node,radiusZero) >= 0.0d0           &
         & ) then
(and remove the subsequent calculation of  `tidalPull` on the next line.

Does that make sense?

Reply:
Yes, that makes sense. I will make the change as you suggested.